src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessContains.cs

```diff
@@ -37,7 +40,19 @@ public void Process(ContainsResultOperator resultOperator, QueryModelVisitor que
 					tree.AddWhereClause(tree.TreeBuilder.Equality(
 						tree.TreeBuilder.Ident(GetFromAlias(tree.Root).AstNode.Text),
 						itemExpression));
-					tree.SetRoot(tree.TreeBuilder.Exists((HqlQuery)tree.Root));
+					if (tree.IsRoot)
+					{
+						tree.AddTakeClause(tree.TreeBuilder.Constant(1));
+						Expression<Func<IEnumerable<object>, bool>> x = l => l.Any();
+						tree.AddListTransformer(x);
+
+						Expression<Func<IEnumerable<bool>, bool>> px = l => l.Any(r => r);
+						tree.AddPostExecuteTransformer(px);
+					}
+					else
+					{
+						tree.SetRoot(tree.TreeBuilder.Exists((HqlQuery) tree.Root));
+					}
```

Comment:
Looks like it comes from https://github.com/nhibernate/nhibernate-core/blob/f36baa502a7f9d21e334e031b9de2b7a334e08bb/src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessAny.cs#L14-L28

So maybe extract code to static method and reuse (something like `ProcessAny.Process(tree)`)?

Reply:
I agree, done.